include/hier/ChildNode.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <ecs/EntityTraits.h>
+
+namespace arch::hier {
+
+struct HierarchyNode;
+
+/// @brief Base class of HierarchyNode, also type of sentinel.
+struct ChildNode {
+	/// @brief Previous sibling
+	ChildNode* _prev;
```

Comment:
In ECS, I think, there shouldn't be any pointers to components that are not local.
It's because ECS should be able to rearrange components, making all stored pointers invalid.

Reply:
ChildNode is only a base class for HierarchyNode, which itself is marked as inPlaceComponent, meaning pointers can only be invalidated by removeComponent